---
title: "msdata2 package"
author: "Chong Tang"
output: [html_document, pdf_document]
bibliography: bibliography.bib
csl: bioinformatics.csl
vignette: >
  %\VignetteIndexEntry{msdata2}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
# Introduction
The `msdata2` package mainly contains a set of public accessible identification/quantitative proteomics datasets. The purpose of this data package is to provide cleaned, formatted, compressed and extensible proteomics datasets as [`MSnSet`](https://www.rdocumentation.org/packages/MSnbase/versions/1.20.7/topics/MSnSet-class) instances to the research community. Moreover, this package will also provide standard and curated datasets to facilitate the benchmarking of proteomics workflows [@Rost2014vh]. In the future, we will also include large raw MS datasets, making use of the [`ExperimentHub`](http://bioconductor.org/packages/release/bioc/html/ExperimentHub.html) cloud infrastructure.

# Available datasets
The package contains the following datasets:    
    I. Raw MS datasets:    
    II. Expression Results:  
    
        -  Label-free quantification:  
            *  Yeast + UPS1 spiked datasets X8  
            *  SWATH-MS Gold Standard Dataset 
            
To list all the datasets in the package, use `msdata2()`.  

# SWATH-MS Datasets
In this vignette, I will use one of the datasets - [`SGS GOLD STANDARD DATASET`](http://compms.org/resources/reference-data/50)  (termed SGS for SWATH-MS
Gold Standard) as the example, to demonstrate SWATH-MS datasets in `msdata2`. The `SGS Dataset` was manually created to validate and benchmark SWATH-MS data analysis algorithms, which consists of known composition [@Rost2014vh].  

### Data Description
The SWATH-MS Gold Standard (SGS) dataset consists of 90 SWATH-MS runs of 422 synthetic stable isotope-labeled standard (SIS) peptides in ten two-fold dilution steps, spiked into three protein backgrounds of varying complexity (water, yeast and human), acquired in three technical replicates. The SGS dataset was manually annotated, resulting in 342 identified and quantified peptides with three or four transitions each. In total, 30,780 chromatograms were inspected and 18,785 were annotated with one true peak group, whereas in 11,995 cases no peak was detected. See also http://www.openswath.org/openswath_data.html for details.  


### Experimental Design
To create SGS Gold Standard Dataset, 422 stable isotope-labeled standard (SIS) peptides (AAA-quantified, Sigma or Thermo) were pooled in a master sample of equal concentration. Then 10 two-fold dilution steps (1, 2, 4, 8, .., 512 times) of the master sample were conducted, resulting in a 512-fold concentration range from 50 fmol/\(\mu\)L to 0.097 fmol/\(\mu\)L. 15 \(\mu\)L of each sample were spiked into 7.5\(\mu\)L of Hela cell lysate (human background), _Saccharomyces cerevisiae_ BY4741 (yeast background) or water (no background). All samples were finally supplemented with 2.5 \(\mu\)L of reference peptides (iRT-Kit, Biognosys AG) for retention time re-alignment, yielding a final sample volume of 25 \(\mu\)L. For each replicate, 2 \(\mu\)L of the samples were injected, yielding a final protein amount of 1 \(\mu\)L of Hela and yeast cell lysate, respectively, loaded on the HPLC column and SIS peptide amounts ranging from 60 fmol to 0.117 fmol. To prepare the spectral library, the most concentrated SIS peptide sample in water was measured in DDA mode.

All samples were analyzed on an Eksigent nanoLC (AS-2/2D plut) system coupled with a SWATH-MS-enabled AB SCIEX TripleTOF 5600 System. For SWATH-MS data analysis, the mass spectrometer was operated using 32 __precursor isolation window__, each of 25 Da effective isolation width (with an addtitional 1 Da overlap on the left side of the window), starting at 400 m/z and ending at 1200 m/z. One full scan is recorded at the beginning for each cycle. Then by applying an acquisition time of 100 ms per scan for MS2, a total cycle time of ~3.3 s is achieved (399-425 m/z precursor isolation window for the first scan, 424-450 m/z for the second ... 1174-1200 m/z for the last scan). Each background has 3 experimental repeats, which leads to 90 different SWATH-MS runs [@Gillet2012vh].


# Importing data
## Load data
```{r, include = TRUE, results = "hide", warning = FALSE, message = FALSE}
library(msdata2)
```
## Exploring the data
Using building function `msdata2()`, we can list all datasets in the package: 
```{r, include=TRUE, warning = FALSE, message = FALSE}
msdata2()
```
We can also fetch the description of each dataset:
```{r, include=TRUE, warning = FALSE, message = FALSE}
data("Rost2014sgsHuman")
?Rost2014sgsHuman
```
How many proteins have been identified by our experiment (e.g. Human background)?
```{r, include=TRUE, warning = FALSE, message = FALSE}
unique(fData(Rost2014sgsHuman)$ProteinName)
```
The SGS datasets were created by stepwise dilutions. Can we also get to know the experimental workflow? For each sample, there are 10 dilution steps, each with 3 replicates, which corresponds to 30 different runs in total.  
```{r, include = TRUE, results = "asis"}
pander::pandoc.table(pData(Rost2014sgsHuman)[1:30,2:4], row.names = F)
```

### Data Visualization
We can also explore the dataset structure with the help of accessible data visualization packages, and also the utilities given by `MSnbase`. I will use SGS dataset with human protein background to illustrate the functions.  
```{r, include = TRUE}
library(ggplot2)
library(reshape2)
library(ggforce)
```

Missingness plot in the expression data:
```{r, include=TRUE, fig.show='asis',fig.width=8, fig.height=8, warning = FALSE, message = FALSE}
naplot(Rost2014sgsHuman)
```
According to the plot given by `naplot` from `MSnbase` package, there're only 5 missing entries in our expression matrix. All the 5 missing values are belonging to the same peptide "HLDSSHPR".  

#### Identification Accuracy
To compute the identification accuracy, all results reported by mProphet above a certain cutoff were taken and it was counted how many of these results were false positive (no manual annotation present) or mis-identified (manual annotation present but at a different retention time, in our case further than 30 seconds away).

Missingness in Expression data:
```{r, include=TRUE}
  table(rowSums(is.na(exprs(Rost2014sgsHuman))))
```

Peptide pattern among 3 BioReplicates:
```{r, include=TRUE, fig.show='asis',fig.width=8, warning = FALSE, message = FALSE}
plot.e <- exprs(Rost2014sgsHuman)
colnames(plot.e) <- pData(Rost2014sgsHuman)$Run
plot.e <- as.data.frame(t(plot.e))
plot.e$Run <- as.numeric(pData(Rost2014sgsHuman)$Run)

plot.run <- melt(plot.e, id.vars = 'Run', variable.name = 'Peptide')
## Add "BioReplicate" column
plot.run$BioReplicate <- cut(plot.run$Run,
                             breaks = c(0, 10, 20, 30),
                             labels = c(1,2,3))
plot.run$Condition <- rep(1:10,3)

##
Rep.labs <- c(`1`="BioReplicate 1", `2` = "BioReplicate 2", `3` = "BioReplicate 3")
p <-ggplot(plot.run, aes(x = Condition, y = value, fill=BioReplicate) ) +
                scale_x_continuous(breaks = seq(1:10)) +
                geom_smooth(method = lm, formula = y ~ splines::bs(x, 3), se = FALSE, show.legend = FALSE, color="darkred") +
                geom_boxplot(aes(group = Run), alpha = 0.5) +
                facet_grid(.~BioReplicate, labeller = as_labeller(Rep.labs)) +
                theme(strip.text.x = element_text(size=9, color="black", face="bold"))
p
```

Correlation between different runs:
```{r, include=TRUE, fig.show='asis',fig.width=8, fig.height=5, warning = FALSE, message = FALSE}
pairs(exprs(Rost2014sgsHuman)[, c(1, 10, 11, 20, 21, 30)]) 
```


##### Density plots
```{r, include=TRUE, fig.show='asis',fig.width=8, fig.height= 7, warning = FALSE, message = FALSE}
library(limma)
limma::plotDensities(exprs(Rost2014sgsHuman)[, c(1, 10, 11, 20, 21, 30)], legend = "topright") 
```

Peptide Count for each Protein
```{r, include=TRUE, fig.show='asis',fig.width=8, fig.height=8, warning = FALSE, message = FALSE}
df.pro <- as.data.frame(table(fData(Rost2014sgsHuman)$ProteinName))
colnames(df.pro) <- c('Protein', 'Freq')
df.pro$Protein <- gsub(".*_", "", df.pro$Protein)
p.pro <- ggplot(df.pro, aes(x=Protein, y = Freq)) + 
                  geom_bar(stat="identity")
p.pro + coord_flip()
```

#### PCA plot
Performs a principal components analysis on the given expression dataset.
```{r, include=TRUE, results = "hide", fig.width=8, fig.height=8, warning = FALSE, message = FALSE}
library(factoextra)
combineFeatures(filterNA(Rost2014sgsHuman), fcol = "ProteinName", method = "robust")
fData(Rost2014sgsHuman)$prot <- fData(Rost2014sgsHuman)$ProteinName
prot <- combineFeatures(filterNA(Rost2014sgsHuman), fcol = "prot", method = "robust")
p2 <- prcomp(t(exprs(prot)), scale = TRUE, center = TRUE)
fviz_pca_ind(p2, habillage = Rost2014sgsHuman$BioReplicate)
```

```{r, include=TRUE, fig.width=8, fig.height=8, warning = FALSE, message = FALSE}
fviz_pca_ind(p2, habillage = Rost2014sgsHuman$BioReplicate, geom="point",
             addEllipses=TRUE, ellipse.level=0.95)
```

##
```{r, include=TRUE, fig.width=8, fig.height=8, warning = FALSE, message = FALSE}
fviz_pca_ind(p2, habillage = Rost2014sgsHuman$Condition, geom="point",
             addEllipses=TRUE, ellipse.level=0.95, palette =  "ucscgb")
```

# Contribution guidelines


# References

